src/l3-resources/atlas-bootstrap/index.ts

```diff
@@ -0,0 +1,129 @@
+  SecretValue,
+  Duration,
+} from "aws-cdk-lib";
+import { Construct } from "constructs";
+
+export const AtlasBasicResources: string[] = [
+  "Cluster",
+  "Project",
+  "DatabaseUser",
+  "ProjectIpAccessList",
+];
+
+export class MongoAtlasBootstrapProps {
+  /**
+   * The IAM role name for CloudFormation Extension Execution.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/registry-public.html
+   * @default auto generat the name.
+   */
+  readonly roleName!: string;
+  /**
+   * The secret profile name for MongoDB Atlas.
+   * @default generate a dummy secret.
+   * @see https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master#2-configure-your-profile
+   */
+  readonly secretProfile!: string;
+  /**
+   * List of strings representing mongoDB atlas types to activate.
+   * the exported AtlasBasicResources can be used for AtlasBasic resources
+   * if not provided non resources will be activated
+   * @example typesToActivate=["Cluster","Project"] this will actiate MongoDB::Atlas::Project && MongoDB::Atlas::Cluster
+   */
+  readonly typesToActivate?: string[];
+}
+
+/**
+ * Generate the CFN extension execution role.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/registry-public.html
+ */
+export class MongoAtlasBootstrap extends Construct {
+  readonly role: iam.IRole;
+  constructor(scope: Construct, id: string, props?: MongoAtlasBootstrapProps) {
+    super(scope, id);
+
+    this.role = new iam.Role(this, "mongoDBAtlasCFNExecutionRole", {
+      maxSessionDuration: Duration.seconds(8400),
+      assumedBy: new iam.CompositePrincipal(
+        new iam.ServicePrincipal("cloudformation.amazonaws.com"),
+        new iam.ServicePrincipal("resources.cloudformation.amazonaws.com"),
+        new iam.ServicePrincipal("lambda.amazonaws.com")
+      ),
+      path: "/",
+      roleName: props?.roleName,
+    });
+
+    const atlasCdkPolicy = new iam.Policy(this, "AtlasCDKExecutionPolicy", {
+      policyName: "AtlasCDKExecution", // Custom Policy Name
+      statements: [
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:DescribeSecret",
+            "secretsmanager:GetSecretValue",
+            "ec2:CreateVpcEndpoint",
+            "ec2:DeleteVpcEndpoints",
+            "cloudformation:CreateResource",
+            "cloudformation:DeleteResource",
+            "cloudformation:GetResource",
+            "cloudformation:GetResourceRequestStatus",
+            "cloudformation:ListResources",
+            "cloudformation:UpdateResource",
+            "iam:AttachRolePolicy",
+            "iam:CreateRole",
+            "iam:DeleteRole",
+            "iam:GetRole",
+            "iam:GetRolePolicy",
+            "iam:ListAttachedRolePolicies",
+            "iam:ListRolePolicies",
+            "iam:PutRolePolicy",
+          ],
+          resources: ["*"],
+        }),
+      ],
+    });
+
```

Comment:
We probably still need to scope down with this as described in the doc using [addSourceAccountCondition()](https://github.com/aws/aws-cdk/blob/c445b8cc6e20d17e4a536f17262646b291a0fe36/packages/aws-cdk-lib/aws-iam/lib/policy-statement.ts#L422).

```
"StringEquals":{
                    "aws:SourceAccount":"123456789012"
                },
```